politeiawww/www.go

```diff
@@ -608,6 +608,50 @@ func (p *politeiawww) handleResetPassword(w http.ResponseWriter, r *http.Request
 	util.RespondWithJSON(w, http.StatusOK, rpr)
 }
 
+// handleProposalAccessTime returns the proposal access time for a given user
+func (p *politeiawww) handleProposalAccessTime(w http.ResponseWriter, r *http.Request) {
+	log.Tracef("handleProposalAccessTime")
+	user, err := p.getSessionUser(w, r)
+	if err != nil {
+		RespondWithError(w, r, 0,
+			"handleProposalAccessTime: getSessionUser %v", err)
+		return
+	}
+	reply, err := p.backend.ProcessProposalAccessTime(user.Email)
+	if err != nil {
+		RespondWithError(w, r, 0,
+			"handleProposalAccessTime: proposalAccessTime %v", err)
+	}
+	util.RespondWithJSON(w, http.StatusOK, reply)
+}
+
+// handleNewProposalAccessTime returns the proposal access time for a user
+func (p *politeiawww) handleNewProposalAccessTime(w http.ResponseWriter, r *http.Request) {
+	log.Tracef("handleNewProposalAccessTime")
+	var pat v1.ProposalAccessTime
+	decoder := json.NewDecoder(r.Body)
```

Comment:
Why are you trying to decode the request body when this request should not have a body associated with it?

The user email value you're getting through the session and the proposal token should be gotten from the URL. Refer to other route handlers on how to grab the token value from the URL.

Reply:
Just to give you a better sense of what I'm saying, this is how this function should look like:
```Go
func (p *politeiawww) handleNewProposalAccessTime(w http.ResponseWriter, r *http.Request) {
	log.Tracef("handleNewProposalAccessTime")

	pathParams := mux.Vars(r)
	token := pathParams["token"]


	user, err := p.getSessionUser(w, r)
	if err != nil {
		RespondWithError(w, r, 0,
			"handleProposalAccessTime: getSessionUser %v", err)
		return
	}
	log.Tracef("%v", user.Email)

	reply, err = p.backend.ProcessNewProposalAccessTime(user.Email, token)
	if err != nil {
		RespondWithError(w, r, 0,
			"handleProposalAccessTime: proposalAccessTime %v", err)
	}
	util.RespondWithJSON(w, http.StatusOK, reply)
}
```
Where reply is the ProposalAccesTime created